### PR TITLE
Iam enablement

### PR DESCRIPTION
**Pull request by eweiter**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,41 @@ var ow = openwhisk(options);
...
+var openwhisk = require('openwhisk');
+var options = {
+                 apihost: 'openwhisk.ng.bluemix.net',
+                 authHandler: authHandler
+              }
+var ow = openwhisk(options)
+ow.actions.invoke('sample').then(result => console.log(result))
+``` 
+
+### Example Using a Authentication Handler with IBM IAM
```
Comment (jthomas) [inline README.md]: Open-source OpenWhisk project documentation does not need to contain provider specific information? This should go in provider documentation instead. Can we remove this section?

File: README.md [hunk 2]
```diff
@@ -64,6 +64,41 @@ var ow = openwhisk(options);
 ow.actions.invoke('sample').then(result => console.log(result))
 ```
 
+### Using 3rd party Authentication Handler
```
Comment (jthomas) [inline README.md]: Change this to a `####` header with lower-case title to be consistent with the overall section style?

File: lib/client.js [hunk 3]
```diff
@@ -91,13 +92,13 @@ class Client {
       apigwSpaceGuid = apiKey.split(':')[0]
     }
 
-    if (!apiKey) {
-      throw new Error(`${messages.INVALID_OPTIONS_ERROR} Missing api_key parameter.`)
+    if (!apiKey && !options.authHandler) {
```
Comment (jthomas) [inline lib/client.js]: SDK uses underscores rather than camelcase for options. Need to consistent with new changes. Change parameter to `auth_handler`?

**Comment by jthomas** [comment]
This is a necessary change to the library to support IAM. The implementation looks good apart from the comments above. 👍

However, this PR was opened with no description or background on the changes. Ideally it should be tied to an existing issue so the community can discuss the changes and debate any API changes. It is hard to merge PRs with no context and doesn't help others provide feedback before an implentation.

**Comment by csantanapr** [comment]
I think there was a mixup with this PR, I will open a new PR with issue later
Closing for now.

**Pull request closed without merging**